libs/shared/src/shared.module.ts

```diff
@@ -1,8 +1,47 @@
-import { Module } from '@nestjs/common';
+import { DynamicModule, Module } from '@nestjs/common';
 import { SharedService } from './shared.service';
+import { ConfigModule, ConfigService } from '@nestjs/config';
+import { ClientProxy, Transport } from '@nestjs/microservices';
+import { ClientProxyFactory } from '@nestjs/microservices';
 
+import { AuthGuard } from './auth.gurad';
 @Module({
-  providers: [SharedService],
-  exports: [SharedService],
+  imports: [
+    ConfigModule.forRoot({
+      isGlobal: true,
+      envFilePath: './.env',
+    }),
+  ],
+  providers: [SharedService, AuthGuard],
+  exports: [SharedService, AuthGuard],
 })
-export class SharedModule {}
+export class SharedModule {
+  static registerRmq(service: string, queue: string): DynamicModule {
+    const serviceProvider = {
+      provide: service,
+      useFactory: (configService: ConfigService): ClientProxy => {
+        const USER = configService.get<string>('RABBITMQ_USER');
+        const PASSWORD = configService.get<string>('RABBITMQ_PASSWORD');
+        const HOST = configService.get<string>('RABBITMQ_HOST');
+
+        return ClientProxyFactory.create({
+          transport: Transport.RMQ,
+          options: {
+            urls: [`amqp://${USER}:${PASSWORD}@${HOST}`],
+            queue: queue,
+            queueOptions: {
+              durable: true,
+            },
+          },
+        });
+      },
+      inject: [ConfigService],
+    };
+
+    return {
+      module: SharedModule,
+      providers: [serviceProvider],
+      exports: [serviceProvider],
+    };
+  }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider refactoring long method for clarity</b></div>
<div id="fix">

Consider breaking down the `registerRmq` method into smaller, more focused functions. This could improve readability and maintainability. For example, you might extract the RabbitMQ configuration logic into a separate method.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -19,28 +19,33 @@
  static registerRmq(service: string, queue: string): DynamicModule {
      const serviceProvider = {
        provide: service,
 -      useFactory: (configService: ConfigService): ClientProxy => {
 -        const USER = configService.get<string>('RABBITMQ_USER');
 -        const PASSWORD = configService.get<string>('RABBITMQ_PASSWORD');
 -        const HOST = configService.get<string>('RABBITMQ_HOST');
 -
 -        return ClientProxyFactory.create({
 -          transport: Transport.RMQ,
 -          options: {
 -            urls: [`amqp://${USER}:${PASSWORD}@${HOST}`],
 -            queue: queue,
 -            queueOptions: {
 -              durable: true,
 -            },
 -          },
 -        });
 -      },
 +      useFactory: (configService: ConfigService): ClientProxy => {
 +        return this.createRmqClient(configService, queue);
 +      },
        inject: [ConfigService],
      };
 
      return {
        module: SharedModule,
        providers: [serviceProvider],
        exports: [serviceProvider],
      };
    }
 +
 +  private static createRmqClient(configService: ConfigService, queue: string): ClientProxy {
 +    const USER = configService.get<string>('RABBITMQ_USER');
 +    const PASSWORD = configService.get<string>('RABBITMQ_PASSWORD');
 +    const HOST = configService.get<string>('RABBITMQ_HOST');
 +
 +    return ClientProxyFactory.create({
 +      transport: Transport.RMQ,
 +      options: {
 +        urls: [`amqp://${USER}:${PASSWORD}@${HOST}`],
 +        queue: queue,
 +        queueOptions: {
 +          durable: true,
 +        },
 +      },
 +    });
 +  }

```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/messenger-api/pull/6#issuecomment-2448418554>#48ef5b</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged